scipy/stats/tests/test_stats.py

```diff
@@ -7947,54 +7947,59 @@ def test_no_args_gh20661(self):
             stats.kruskal()
 
 
+@array_api_compatible
 class TestCombinePvalues:
 
-    def test_fisher(self):
+    def test_fisher(self, xp):
         # Example taken from https://en.wikipedia.org/wiki/Fisher%27s_exact_test#Example
-        xsq, p = stats.combine_pvalues([.01, .2, .3], method='fisher')
-        assert_approx_equal(p, 0.02156, significant=4)
-
-    def test_stouffer(self):
-        Z, p = stats.combine_pvalues([.01, .2, .3], method='stouffer')
-        assert_approx_equal(p, 0.01651, significant=4)
-
-    def test_stouffer2(self):
-        Z, p = stats.combine_pvalues([.5, .5, .5], method='stouffer')
-        assert_approx_equal(p, 0.5, significant=4)
-
-    def test_weighted_stouffer(self):
-        Z, p = stats.combine_pvalues([.01, .2, .3], method='stouffer',
-                                     weights=np.ones(3))
-        assert_approx_equal(p, 0.01651, significant=4)
-
-    def test_weighted_stouffer2(self):
-        Z, p = stats.combine_pvalues([.01, .2, .3], method='stouffer',
-                                     weights=np.array((1, 4, 9)))
-        assert_approx_equal(p, 0.1464, significant=4)
-
-    def test_pearson(self):
-        Z, p = stats.combine_pvalues([.01, .2, .3], method='pearson')
-        assert_approx_equal(p, 0.02213, significant=4)
-
-    def test_tippett(self):
-        Z, p = stats.combine_pvalues([.01, .2, .3], method='tippett')
-        assert_approx_equal(p, 0.0297, significant=4)
-
-    def test_mudholkar_george(self):
-        Z, p = stats.combine_pvalues([.1, .1, .1], method='mudholkar_george')
-        assert_approx_equal(p, 0.019462, significant=4)
-
-    def test_mudholkar_george_equal_fisher_pearson_average(self):
-        Z, p = stats.combine_pvalues([.01, .2, .3], method='mudholkar_george')
-        Z_f, p_f = stats.combine_pvalues([.01, .2, .3], method='fisher')
-        Z_p, p_p = stats.combine_pvalues([.01, .2, .3], method='pearson')
-        assert_approx_equal(0.5 * (Z_f+Z_p), Z, significant=4)
+        xsq, p = stats.combine_pvalues(xp.asarray([.01, .2, .3]), method='fisher')
+        xp_assert_close(p, xp.asarray(0.02156), rtol=1e-4)
+
+    def test_stouffer(self, xp):
+        Z, p = stats.combine_pvalues(xp.asarray([.01, .2, .3]), method='stouffer')
+        xp_assert_close(p, xp.asarray(0.01651), rtol=1e-3)
```

Comment:
I'd be surprised if you can't use `rtol` equal to the precision of the reference p-value, since it is always much coarser than the precision of the floating point dtype.

Reply:
Tightening the tolerance causes a failure on all backends
![image](https://github.com/scipy/scipy/assets/60778417/ea5f536f-c232-4a61-9926-ee209b76647c)

Reply:
Them the reference p-value is probably wrong. Fine to leave it alone for this PR, but maybe when this is refactored I'll look for a better reference that we can cite in the test.

Reply:
Well this is indeed what R gives.
```python3
library(metap)
x = c(0.01, 0.2, 0.3)
sumz(x)
```
I'll see if there are any hints in the code or documentation as to why there is such a discrepancy.

Reply:
No such hints. I think It's worth follow-up in a refactoring PR. I don't think an issue is needed if you were willing to submit such a PR soon. LMK either way.